### DISCUSSION TITLE
store object in value

### DISCUSSION THREAD

**Discussion by ericksonholguin** [react-hook-form/react-hook-form#3414] [Q&A]
Hello, I have loved this library since compared to the others, this one takes you a great journey in advance, I start with this little story, it is because I have come across a topic and it is to be able to store objects in value, no I want to use strange tricks or anything like that, I was using their library with React Select, but I realized that React Select did not convince me for many reasons and therefore I created my own Select component, the problem is that it takes an object with different fields, such as value, label, position, etc. 

So, how could I show a field of these as a label and when submitting the form, bring me the complete selected object? Since I was looking in the previous problems, I saw that a person asked before, but the answer that you gave it a codesandbox and it is no longer available, thank you very much, I hope you can help me

**Comment by bluebill1049** [comment]
I think you can wrap your component with Controller to solve that issue, you can expose any object as you want.

**Reply by ericksonholguin** [reply]
I knew that more or less that was the way, but I can not achieve my mission, I have been trying for a long time, this is an image of the many failed attempts

![image](https://user-images.githubusercontent.com/38663299/98743508-09527500-2365-11eb-985d-b2d97199d8b5.png)

It is not the most elegant code, but since I am in test mode and that, my intention is not that you understand it, rather it is that you make a small example and publish it, please, I am super behind in my project

**Reply by bluebill1049** [reply]
have a look at this example in the example folder: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc

`<ParseTextarea />` is what you want to do, build a controlled input and expose `value`, `onChange`, `onBlur` and let the `Controller` to collect and validate the value itself.

**Reply by ericksonholguin** [reply]
Look brother, in this example that you gave me when I submit, it does not return the object as such, rather it returns the value field, I would like that instead of returning "**ReactSelect:" chocolate "**", it returns **ReactSelect: {
value: "chocolate",
label: "chocolate"
}**

![image](https://user-images.githubusercontent.com/38663299/98746116-344fbf80-238b-11eb-93ae-8d92576464a7.png)

**Reply by bluebill1049** [reply]
by default react-select's value is an `object`,  look at line 66, that's what's the value get to send to react hook form, you can send other value if you want, as long as your value is the correct shape for react-select.

**Reply by bluebill1049** [reply]
if you can change line 65

`onChange={data => props.onChange(data)} value={value)` then you get the entire object

**Reply by ericksonholguin** [reply]
Thank, Ya profundizaré en el tema, pero parece funcionar
